README.md

```diff
@@ -7,20 +7,136 @@ Initially generated with [xmlpg](https://github.com/open-dis/xmlpg).
+* SDL2 and SDL2_net are required libraries to compile the examples.
+  These can usually be install via Linux package managers.
+  More details, and instructions for Windows are included [below](#SDL2-and-SDL2_net-Install-Instructions).
 
-### Linux
+  **NOTE:** Windows users will need to either add the DLL folders to their path, or copy the DLLs to the output directory (`Debug`).
 
+### Linux / GNU Makefiles
+
+1. Run `mkdir build`
+1. Run `cd build`
+1. Run `cmake ..`
+  1. Optionally, `-DCMAKE_INSTALL_PREFIX=<custom-path-to-install>` to set a custom directory to install the bin, include, and lib output directories.
+1. Run `make` - this will output the libOpenDIS6.so, and libOpenDIS7.so libraries in the build directory along with the Example Applications.
+1. The below steps are optional
+  1. Run `make package` to build Linux package files. Currently this will produce a Red Hat RPM package, Debian DEB package, and 2 compressed tarball (XZ, GZ).
+  1. Run `make install` to install bin, lib, and dir, into `CMAKE_INSTALL_PREFIX`
+     WARNING: `CMAKE_INSTALL_PREFIX` default can to somewhere `/usr/local/`, if not specified with the `-D` flag as shown in Step 3.1.
+     If you're unsure where to install, and want to keep your `/usr/local/` directory clean, run `cmake .. -DCMAKE_INSTALL_PREFIX=./install`.
+     This will cause `make install` to create a local install directory, from which you can move files elsewhere at a later date.
+
+#### Windows with Visual Studio
+1. Open `CMake (cmake-gui)` via the the start menu.
+2. Enter the open-dis-cpp directory path into the Source and Build fields.
+3. Click the `+ Add Entry` button and enter the following details:
+  Name: `SDL_INC_DIR`
+  Type: `PATH`
+  Value: `<SDL2-install-folder>/include`
+  (if you follow the below [SDL2 Windows Install Instruction](#Windows-Install-Instructions) this will be `C:/SDL2/include`)
+3. Click the `+ Add Entry` button and enter the following details:
+  Name: `SDL_LIB_DIR`
+  Type: `PATH`
+  Value: `<SDL2-install-folder>/lib/x64` (64 bit) or `<SDL2-install-folder>/lib/x86` (32 bit)
+  (if you follow the below [SDL2 Windows Install Instruction](#Windows-Install-Instructions) this will be `C:/SDL2/lib/x64`)
+4. Click Configure and follow the prompts, selecting the correct generator (i.e. Visual Studio version).
+5. Click Generate
+6. Click Open Project - This should open the generates solution file in Visual Studio
+7. Build the Solution (`Ctrl + Shift + B`)
+
+These steps were tested with Visual Studio 16 2019 (Community Edition).
+Currently, only OpenDIS 6 and the Example Applications compile.
+The library and executable files are output to a `Debug` directory.
+
+To run the executables, either the value of the `SDL_LIB_DIR` variable should be added to your path,
+or `SDL2.dll` and `SDL2_net.dll` need to be copied to the `Debug` directory.
+
+### Cleaning CMake files
+
+To quickly clean up CMake output files, use `git clean -xdf`.
```

Comment:
If you do not put the build folder into your source, you will not have this problem.

Reply:
I found the `./build` a good convention for Linux, but from memory when I did that in Visual Studio (VS) the source files were treated as external - grouped in the solution explorer with stdio files, and SDL2.
I assumed this was because the files were below the current directory.
The Windows VS instruction use the same source and build instructions to avoid the grouping, so the clean-up method is mainly for Windows.
Overall, the instructions are meant to be a simple guide to get start, assuming devs with more CMake experience know how to customise their build.